src/mongo_odm/macros.hpp

```diff
@@ -0,0 +1,97 @@
+// Copyright 2016 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <mongo_odm/config/prelude.hpp>
+
+// Preprocessor templates for manipulating multiple arguments.
+#define MONGO_ODM_PP_NARG(...) MONGO_ODM_PP_NARG_(__VA_ARGS__, MONGO_ODM_PP_RSEQ_N())
+#define MONGO_ODM_PP_NARG_(...) MONGO_ODM_PP_ARG_N(__VA_ARGS__)
+#define MONGO_ODM_PP_ARG_N(_0, _1, _2, _3, _4, _5, _6, _7, _8, _9, _10, _11, _12, _13, _14, _15, \
```

Comment:
Is this used anywhere other than for `ODM_CHILD{{N}}`? If not, what do you think about putting these `PP_NARG`s marcros into that generated file?